idea/src/org/jetbrains/kotlin/idea/inspections/RedundantVisibilityModifierInspection.kt

```diff
@@ -18,7 +23,18 @@ class RedundantVisibilityModifierInspection : AbstractKotlinInspection(), Cleanu
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean, session: LocalInspectionToolSession): PsiElementVisitor {
         return declarationVisitor { declaration ->
             val visibilityModifier = declaration.visibilityModifier() ?: return@declarationVisitor
-            val implicitVisibility = declaration.implicitVisibility()
+            val implicitVisibility = when {
```

Comment:
I'd say here we should fix `KtDeclaration.implicitVisibility()` itself and not insert some work-arounds just here.